### PR TITLE
Gsnyder/notifications license info and more

### PR DESCRIPTION
**Pull request by gsnyder2007**
Adding some endpoints for license and notifications.

Changed some stuff for getting information about code (scan) locations.

Added some code for checking the job status for jobs related to scans, especially snippet scans.

### CODE REVIEW

File: examples/wait_for_scan_results.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    if response.status_code == 200:
+        project_version_obj = response.json()
+    else:
+        raise Exception("Unable to retrieve project version at URL {}".format(
+            project_version_url))
+
+    version_name = project_version_obj['versionName']
+    project_url = hub.get_link(project_version_obj, "project")
+    response = hub.execute_get(project_url)
+    if response.status_code == 200:
```
Comment (jackeekaplan) [inline examples/wait_for_scan_results.py]: is it possible to have project_version_url return 200 and not the project_url?

File: examples/check_scan_jobs_status.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+            scans_to_check = self.scan_history[-1:]
+
+        assert len(scans_to_check) > 0, "Uh oh, no scans to check. That shouldn't happen."
+        assert len(self.related_jobs) > 0, "Uh oh, no related jobs to check. That shouldn't happen."
+
+        checks = [s['status'].lower() == "complete" for s in scans_to_check] 
+        checks.extend([j['status'].lower() == "completed" for j in self.related_jobs])
+
+        self.failed_jobs = [j for j in self.related_jobs if j['status'].lower() != "completed"]
+        self.failed_scans = [s for s in scans_to_check if s['status'].lower() != "complete"]
```
Comment (jackeekaplan) [inline examples/check_scan_jobs_status.py]: are in-progress jobs considered failed?

File: examples/wait_for_scan_results.py (hunk 1 above)
Comment (gsnyder2007) [reply examples/wait_for_scan_results.py]: Well, it shouldn't be possible. I think you're alluding to the assumption that you can only have a version if there is a project that owns it? And I think your point is I can simplify this code by eliminating some of the checks?

Comment (jackeekaplan) [reply examples/wait_for_scan_results.py]: yes, just wasn't sure if this was something you'd run into that I hadn't yet :)

File: examples/check_scan_jobs_status.py (hunk 2 above)
Comment (gsnyder2007) [reply examples/check_scan_jobs_status.py]: Short answer: no. However, from the POV of this code we are checking for success and anything that is not success is a fail. 

I think it makes sense though to extend the code to differentiate between in-progress and outright fail. I'm worried about the case of a scan/job that is running and will not finish. We've had cases where scans hang and I want to make sure that situation is a 'fail' in the sense that a non-zero exit status is set if/when all jobs and scans have not complete(d). Make sense?

File: examples/wait_for_scan_results.py (hunk 1 above)
Comment (gsnyder2007) [reply examples/wait_for_scan_results.py]: Call it defensive coding style. Sometimes I use assertions in code to denote the _thing that should never happen_ (but sometimes does anyhow). Other times I write, potentially redundant, code to check things. Sometimes I just write bad code - refactoring was invented for people like me! :-)

File: examples/check_scan_jobs_status.py (hunk 2 above)
Comment (jackeekaplan) [reply examples/check_scan_jobs_status.py]: makes sense

**Pull request merged**